### PR TITLE
AK1003 - Add `ReceiveAsync` with sync lambda analyzer and code fix

### LINKED ISSUE #57: AK1000: warn users about using `ReceiveAsync` when not actually doing any `await`-ing
**Is your feature request related to a problem? Please describe.**

From the submission on https://github.com/akkadotnet/akka.analyzers/issues/55

```csharp
public sealed class MyActor : ReceiveActor{
   public MyActor(){
      ReceiveAsync<string>(str => Sender.Tell(str)); // should get a warning here
   }
}
```

Using `ReceiveAsync` this way is less performant than a normal `Receive` because this generates an extra [`ActorTask` message](Akka.Dispatch.SysMsg.ActorTask) plus an additional suspend + resume of the actor's mailbox. That overhead is fine if you're `await`-ing something, but if the code is essentially synchronous then we should not do so.

**Describe the solution you'd like**

Issue a `WARNING` level severity here and a code fix to rewrite the `ReceiveAsync` to a regular `Receive` if the `ReceiveAsync` doesn't actually have any `await`s happening inside of it.

**Additional context**

If the user is invoking a declared method as the `ReceiveAsync` handler instead of an anonymous delegate, that might be a lot trickier to analyze correctly as the method can get used across multiple callsites. Maybe it's best to skip doing that for now.

### PR DESCRIPTION
**Pull request by Arkatufus**
Partially fixes #57

## Changes

Add analyzer and code fix for `ReceiveAsync<T>()` and `ReceiveAnyAsync()` invocations that does not use async lambda expression.

## Notes
This implementation does not analyze `ReceiveAsync<T>()` and `ReceiveAnyAsync()` that uses a method delegate. This will need a separate analyzer.

Example:
```csharp
public sealed class MyActor : ReceiveActor
{
    public MyActor()
    {
        ReceiveAsync<string>(Handler); // This can't be analyzed, for now.
    }
    
    private async Task Handler(string s) { }
}
```

### ReceiveAsync with async keyword and no await keywords

```csharp
ReceiveAsync<string>(async str =>
{
    Sender.Tell(str);
});
```
to
```csharp
Receive<string>(str =>
{
    Sender.Tell(str);
});
```

### ReceiveAnyAsync with async keyword and no await keywords

```csharp
ReceiveAnyAsync(async o =>
{
    Sender.Tell(o);
});
```
to
```csharp
ReceiveAny(o =>
{
    Sender.Tell(o);
});
```

### ReceiveAsync with no body block with async keyword and no await keywords

```csharp
ReceiveAsync<string>(async str => Sender.Tell(str));
```
to
```csharp
Receive<string>(str => Sender.Tell(str));
```

### ReceiveAnyAsync with no body block with async keyword and no await keywords

```csharp
ReceiveAnyAsync(async o => Sender.Tell(o));
```
to
```csharp
ReceiveAny(o => Sender.Tell(o));
```

### CODE REVIEW

**Review by Aaronontheweb — COMMENTED** [review]
Just left one question for you @Arkatufus

File: src/Akka.Analyzers/AK1000/ShouldNotUseReceiveAsyncWithoutAsyncLambdaAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                    arg.Expression is LambdaExpressionSyntax expr 
+                    && IsFuncOfTTask(semanticModel.GetTypeInfo(expr).ConvertedType));
+
+            if (lambdaArg is null)
+                return;
+
+            var lambdaExpr = (LambdaExpressionSyntax)lambdaArg.Expression;
+            
+            // first case, lambda expression is not prefixed with the `async` keyword
+            if (!lambdaExpr.AsyncKeyword.IsKind(SyntaxKind.AsyncKeyword))
```
[the remark this answers is not in the document]
Comment (Arkatufus) [reply src/Akka.Analyzers/AK1000/ShouldNotUseReceiveAsyncWithoutAsyncLambdaAnalyzer.cs:44]: You're right, in this case, I'd have to expand the analyzer to see if the return value is `Task.CompletedTask` or not, that would signal that the whole method is actually synchronous.

[the remark this answers is not in the document]
Comment (Arkatufus) [reply src/Akka.Analyzers/AK1000/ShouldNotUseReceiveAsyncWithoutAsyncLambdaAnalyzer.cs:44]: On second thought, I'd also have to analyze that the return value isn't a `Task.FromResult` that returns a static value... hmmm...

[the remark this answers is not in the document]
Comment (Aaronontheweb) [reply src/Akka.Analyzers/AK1000/ShouldNotUseReceiveAsyncWithoutAsyncLambdaAnalyzer.cs:44]: Yeah we probably should just ignore that case.

[the remark this answers is not in the document]
Comment (Arkatufus) [reply src/Akka.Analyzers/AK1000/ShouldNotUseReceiveAsyncWithoutAsyncLambdaAnalyzer.cs:45]: Behaviour change, we're ignoring this case now, we're assuming that the user is doing the proper thing of returning an actual async `Task` and not `Task.CompletedTask` or `Task.FromResult`

[the remark this answers is not in the document]
Comment (Aaronontheweb) [reply src/Akka.Analyzers/AK1000/ShouldNotUseReceiveAsyncWithoutAsyncLambdaAnalyzer.cs:45]: LGTM

**Pull request merged**